content/installation/java/configuration/YAML-properties-java.md

```diff
@@ -0,0 +1,203 @@
+1. System property value
+1. Environment variable value
+1. YAML configuration file value
+1. Contrast UI value
+1. Default value
+
+<!-- Other use and file location info. This will be done after CONTRAST-26125  -->
+
+Go to the [Java YAML Template](installation-netconfig.html#net-template) for fully formatted properties that you can copy and use in your own agent configuration files. 
+
+## Set as System Properties
+
+You can also set all of the following YAML properties as system properties. Derive the system property key from the YAML by joining every node with a "." until you reach the bottom property. 
+
+> **Example:** If you want to override the `contrast` property, as given below, you can set `-Dcontrast.enable=false` as a system property.
+  * contrast: 
+    * enable: `true`
+
+
+## Configuration Options
+
+### Contrast UI properties
+
+Use the properties in this section to connect the Java agent to the Contrast UI. The proxy settings allow the agent to communicate with the Contrast UI over a proxy.
+
+
+* **contrast**: 
+
+  * **enable**: Only set this property if you want to turn off Contrast. Set to `true` to turn the agent on; set to `false` to turn the agent off.
+  * **url**: Set the URL for the Contrast UI. <br> Example: https://app.contrastsecurity.com/Contrast. **Required.** 
+  * **api_key**: Set the API key needed to communicate with the Contrast UI. **Required.**
+  * **service_key**: Set the service key needed to communicate with the Contrast UI. It is used to calculate the Authorization header. **Required.**
+  * **user_name**: Set the user name used to communicate with the Contrast UI. It is used to calculate the Authorization header. **Required.**
+
+  * **proxy**:
+    * **enable**: Add a property value to determine if the agent should communicate with the Contrast UI over a proxy. If a property value is not present, the presence of a valid proxy host and port determines enabled status. Value options are `true` or `false`
+    * **host**: Set the proxy host. It must be set with port and scheme. <br> Example: `localhost`
+    * **port**: Set the proxy port. It must be set with host and scheme. <br> Example: `1234`
+    * **scheme**: Set the proxy scheme. It must be set with host and port. <br> Example: `http` or `https`
+    * **url**: Set as an alternate for `scheme://host:port`. It takes precedence over the other settings, if specified; however, an error will be thrown if both the URI and individual properties are set.
+    * **user**: Set the proxy user.
+    * **pass**: Set the proxy password.
+    * **auth_type**: Set the proxy authentication type. Value options are `NTLM`, `Digest`, and `Basic`.
+
+
+### Contrast agent properties
+
+Use the properties in this section to control the way and frequency with which the Java agent communicates to logs and to the Contrast UI.
+If these values are not set, the agent will use the values set in the Contrast UI.
+
```

Comment:
We've added two properties under the `agent` node. Should we add those here? The properties are
```
agent:

  # Set how long to run the agent before shutting down itself (in milliseconds). A negative value disables scheduled shutdown.
  shutdown_time_ms: NEEDS_TO_BE_SET

  # Enable to deinstrumentation classes on shutdown. If this is not enabled, the agent disables sensors on shutdown, but leaves instrumentation.
  deinstrument_on_shutdown: NEEDS_TO_BE_SET
```

Reply:
We'll need to update the description on line 63 too. Maybe something like

> Use the properties in this section to control agent wide behaviors. For instance, this section can be used to control the way in which the agent starts up and shuts down, and the way in which it communicates to logs and to the Contrast UI.